cmd/skopeo/sync.go

```diff
@@ -280,8 +280,9 @@ func imagesToCopyFromRegistry(registryName string, cfg registrySyncConfig, sourc
 	serverCtx.DockerDaemonCertPath = cfg.CertDir
 	serverCtx.DockerDaemonInsecureSkipTLSVerify = (cfg.TLSVerify.skip == types.OptionalBoolTrue)
 	serverCtx.DockerInsecureSkipTLSVerify = cfg.TLSVerify.skip
-	serverCtx.DockerAuthConfig = &cfg.Credentials
-
+	if cfg.Credentials.Username != "" && cfg.Credentials.Password != "" {
```

Comment:
What about `IdentityToken`?

And then probably use `||` — or, to make this future-proof, `cfg.Credentials != types.DockerAuthConfig{}`